examples/ros-service-call-module/src/JsonSchemaForm/IntegerInput.tsx

```diff
@@ -0,0 +1,64 @@
+import { FC, useState, useEffect } from "react";
+import { TextField } from "@formant/ui-sdk";
+
+interface IIntegerInputProps {
+  jsonSchemaObject: any;
+  currentStateObject: any;
+  property: string;
+  defaultValue: string;
+}
+
+export const IntegerInput: FC<IIntegerInputProps> = ({
+  jsonSchemaObject,
+  currentStateObject,
+  property,
+  defaultValue,
+}) => {
+  const [currentValue, setCurrentValue] = useState<number | string>();
+
+  const [error, setError] = useState("");
+
+  useEffect(() => {
+    if (isNaN(parseInt(defaultValue))) return;
+    setCurrentValue(parseInt(defaultValue));
+  }, []);
+  const isValid = (_: string) => {
+    if (_ === "") return true;
+    if (!!_) {
+      const newestInput = _.at(-1);
+      const letterToFloat = parseFloat(newestInput!);
+      if (Number.isInteger(letterToFloat)) return true;
+    }
+    setError("Please enter a valid integer");
+    return false;
+  };
+
+  return (
+    <TextField
+      type="phone"
+      key={jsonSchemaObject.properties[property].title}
+      sx={{ marginBottom: "36px" }}
+      fullWidth={true}
+      value={currentValue}
+      onChange={(ev) => {
+        if (!isValid(ev.target.value)) {
+          return;
+        }
+        setError("");
+        if (ev.target.value === "") {
+          setCurrentValue("");
+        } else {
+          setCurrentValue(parseInt(ev.target.value));
+        }
+        currentStateObject[jsonSchemaObject.properties[property].title] =
+          ev.target.value;
```

Comment:
I'm going to have to parse this _deeper_ but _traditionally_ this would be a react _no-no_, right? A component shouldn't be _assigning/mutating_ to an object that gets passed in via props. So, i'm going to have to brain compile this a little more and understand where this is coming from/going to.